addon/mixins/data-adapter-mixin.js

```diff
@@ -125,7 +125,9 @@ export default Mixin.create({
   },
 
   authorize() {
-    assert('The `authorize` method should be overridden in your application adapter. It should accept a single argument, the request object.');
+    let [ major, _ ] = Ember.VERSION.split(".");
+
+    assert('The `authorize` method should be overridden in your application adapter. It should accept a single argument, the request object.', major >= 2);
```

Comment:
Ember pre 2.0 doesn't have the `headers` property, so this should still be replaced on those versions.

Another option is to use `warn` instead of `assert` to just show a warning to users, instead of throwing an error.

Reply:
I think this would have to check the Ember Data version, not the Ember version?

Reply:
Yes 🤦‍♀